packages/arb-token-bridge-ui/src/components/common/AddCustomChain.tsx

```diff
@@ -0,0 +1,267 @@
+    utils: string
+    validatorWalletCreator: string
+  }
+  tokenBridgeContracts: {
+    l2Contracts: Contracts
+    l3Contracts: Contracts
+  }
+}
+export function getCustomChainsFromLocalStorage(): ChainWithRpcUrl[] {
+  const customChainsFromLocalStorage = localStorage.getItem(localStorageKey)
+
+  if (customChainsFromLocalStorage) {
+    return (JSON.parse(customChainsFromLocalStorage) as ChainWithRpcUrl[])
+      .filter(
+        // filter again in case local storage is compromized
+        chain => !allowedParentChainIds.includes(Number(chain.chainID))
+      )
+      .map(chain => {
+        return {
+          ...chain,
+          // make sure chainID is numeric
+          chainID: Number(chain.chainID)
+        }
+      })
+  }
+
+  return []
+}
+
+function mapOrbitConfigToOrbitChain(data: OrbitConfig): ChainWithRpcUrl {
+  return {
+    chainID: data.chainInfo.chainId,
+    confirmPeriodBlocks: 150,
+    ethBridge: {
+      bridge: data.coreContracts.bridge,
+      inbox: data.coreContracts.inbox,
+      outbox: data.coreContracts.outbox,
+      rollup: data.coreContracts.rollup,
+      sequencerInbox: data.coreContracts.sequencerInbox
+    },
+    rpcUrl: data.chainInfo.rpcUrl,
+    explorerUrl: data.chainInfo.explorerUrl,
+    isCustom: true,
+    name: data.chainInfo.chainName,
+    partnerChainID: data.chainInfo.parentChainId,
+    retryableLifetimeSeconds: 604800,
+    nitroGenesisBlock: 0,
+    nitroGenesisL1Block: 0,
+    depositTimeout: 900000,
+    isArbitrum: true,
+    tokenBridge: {
+      l1CustomGateway: data.tokenBridgeContracts.l2Contracts.customGateway,
+      l1ERC20Gateway: data.tokenBridgeContracts.l2Contracts.standardGateway,
+      l1GatewayRouter: data.tokenBridgeContracts.l2Contracts.router,
+      l1MultiCall: data.tokenBridgeContracts.l2Contracts.multicall,
+      l1ProxyAdmin: data.tokenBridgeContracts.l2Contracts.proxyAdmin,
+      l1Weth: data.tokenBridgeContracts.l2Contracts.weth,
+      l1WethGateway: data.tokenBridgeContracts.l2Contracts.wethGateway,
+      l2CustomGateway: data.tokenBridgeContracts.l3Contracts.customGateway,
+      l2ERC20Gateway: data.tokenBridgeContracts.l3Contracts.standardGateway,
+      l2GatewayRouter: data.tokenBridgeContracts.l3Contracts.router,
+      l2Multicall: data.tokenBridgeContracts.l3Contracts.multicall,
+      l2ProxyAdmin: data.tokenBridgeContracts.l3Contracts.proxyAdmin,
+      l2Weth: data.tokenBridgeContracts.l3Contracts.weth,
+      l2WethGateway: data.tokenBridgeContracts.l3Contracts.wethGateway
+    }
+  }
+}
+
+export const AddCustomChain = () => {
+  const [customChains, setCustomChains] = useState<Chain[]>(
+    getCustomChainsFromLocalStorage()
+  )
+  const [chainJson, setChainJson] = useState<string>('')
+  const [error, setError] = useState<string | null>(null)
+  const [isDeleteConfirmation, setIsDeleteConfirmation] = useState(false)
+
+  function onAddChain() {
+    setError(null)
+
+    try {
+      const data = (
+        chainJson.trim()
```

Comment:
this returns a string, why is it used for a ternary condition?

is there a reason that we don't directly call `JSON.parse(chainJson.replace(/[\r\n]+/g, ''))` here?

Reply:
Must have had some old checks here. Updated